VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Public/VitruvioBlueprintLibrary.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2021 Esri
+ *
+ * Licensed under the Apache License Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "RulePackage.h"
+#include "VitruvioActor.h"
+#include "Kismet/BlueprintFunctionLibrary.h"
+
+#include "VitruvioBlueprintLibrary.generated.h"
+
+UCLASS()
+class UVitruvioBlueprintLibrary : public UBlueprintFunctionLibrary
+
+{
+	GENERATED_BODY()
+
+public:
+	/**
+	 * Returns all Actors attached to the given Actor which are viable to be converted to VitruvioActors.
+	 *
+	 * @param Root the root actor whose children are checked if they are viable Vitruvio Actors.
+	 * @return all Actors attached to the given Actor which are viable to be converted to VitruvioActors.
+	 */
+	UFUNCTION(BlueprintCallable, Category = "Vitruvio")
+	static VITRUVIO_API TArray<AActor*> GetViableVitruvioActorsInHierarchy(AActor* Root);
+
+	/**
+	 * Returns whether the given Actor can be converted to a VitruvioActor (see also ConvertToVitruvioActor). Converting an Actor to a VitruvioActor
+	 * is only possible if the Actor already has a valid initial shape component attached (eg. a StaticMeshComponent).

```

Comment:
We should probably be more precise as to what kind of actors can be converted. (I.e. we can't convert a regular actor with vitruvio component attached to a VitruvioActor)

Reply:
Hm I thought the second sentence explained that? 

"Converting an Actor to a VitruvioActor is only possible if the Actor already has a valid initial shape component attached".

Maybe by adding something like "and does not already have a VitruvioComponent attached"?

Reply:
Sounds good. I was just confused as I thought we could "promote" a regular actor that used a vitruvio component to a proper vitruvio actor. I think the changes you propose would definitely clarify this confusion.

Reply:
Ahh ok! Makes sense then.